src/main/scala/is/hail/io/RowStore.scala

```diff
@@ -521,7 +523,95 @@ object RichRDDRegionValue {
 }
 
 class RichRDDRegionValue(val rdd: RDD[RegionValue]) extends AnyVal {
-  def writeRows(path: String, t: TStruct): Array[Long] = {
+  def writeRows(path: String, t: TStruct): (Array[String], Array[Long]) = {
     rdd.writePartitions(path, RichRDDRegionValue.writeRowsPartition(t))
   }
+
+  def writeRowsSplit(path: String, t: MatrixType, partitioner: OrderedRVDPartitioner): Array[Long] = {
+    val sc = rdd.sparkContext
+    val hConf = sc.hadoopConfiguration
+
+    hConf.mkDir(path + "/rows/rows/parts")
+    hConf.mkDir(path + "/entries/rows/parts")
+
+    val sHConfBc = sc.broadcast(new SerializableHadoopConfiguration(hConf))
+
+    val nPartitions = rdd.getNumPartitions
+    val d = digitsNeeded(nPartitions)
+
+    val partFiles = Array.tabulate[String](nPartitions) { i => partFile(d, i) }
+
+    val fullRowType = t.rvRowType
+    val rowsRVType = t.rowType
+    val localEntriesIndex: Int = t.entriesIdx
+
+    val entriesRVType = TStruct(
+      MatrixType.entriesIdentifier -> TArray(t.entryType))
+
+    val partitionCounts = rdd.mapPartitionsWithIndex { case (i, it) =>
+      val hConf = sHConfBc.value.value
+
+      val f = partFile(d, i)
+
+      val rowsPartPath = path + "/rows/rows/parts/" + f
+      val rowsOS = hConf.unsafeWriter(rowsPartPath)
```

Comment:
this should be the serializable hadoop conf, right?

Reply:
It is?   Just above: `val hConf = sHConfBc.value.value`.